ext/mysql2/result.c

```diff
@@ -78,6 +78,7 @@ static void rb_mysql_result_free_result(mysql2_result_wrapper * wrapper) {
   if (wrapper && wrapper->resultFreed != 1) {
     /* FIXME: this may call flush_use_result, which can hit the socket */
     mysql_free_result(wrapper->result);
+    wrapper->lastRowProcessed = 0;
```

Comment:
Looks like that it will break the `:cache_rows => true` behavior.
https://github.com/ajaska/mysql2/blob/0.3.x/ext/mysql2/result.c#L528-L535